src/aof.c

```diff
@@ -280,22 +280,47 @@ void aofLoadManifestFromDisk(void) {
         }
 
         line = sdstrim(sdsnew(buf), " \t\r\n");
-        argv = sdssplitargs(line, &argc);
-        /* 'argc < 6' was done for forward compatibility. */
-        if (argv == NULL || argc < 6 || (argc % 2)) {
+        if (!sdslen(line)) {
+            err = "The AOF manifest file is invalid format";
+            goto loaderr;
+        }
+
+        /* Because we have to deal with the case where 'appendfilename' 
+         * contains spaces, we have to deal with AOF_MANIFEST_KEY_FILE_NAME
+         * separately. */
+        char *file_pos = strstr(line, AOF_MANIFEST_KEY_FILE_NAME);
+        char *seq_pos = rstrstr(line, AOF_MANIFEST_KEY_FILE_SEQ);
+        if (file_pos == NULL || seq_pos == NULL) {
+            err = "The AOF manifest file is invalid format";
+            goto loaderr;
+        }
+
+        char *filename_pos = file_pos + strlen(AOF_MANIFEST_KEY_FILE_NAME) + 1;
+        sds file_name = sdsnewlen(filename_pos, seq_pos - filename_pos - 1);
+        if (!sdslen(file_name)) {
+            sdsfree(file_name);
+            err = "The AOF manifest file is invalid format";
+            goto loaderr;
+        }
```

Comment:
i have a feeling this is overly complicated, and also i imagine the manifest file would look bad:
```
file bla bla bla seq 1 type b
```
i.e. as a human reader, i'm not sure what i'm looking at...

i think the `sdssplitargs` which we already use, can handle this case:
```
file "bla bla bla" seq 1 type b
```
as well as the old one:
```
file filename.aof seq 1 type b
```
and then all that's left is to add a quote when generating the manifest if the file name contains white space
am i missing anything?

Reply:
@oranagra Great suggestion.

I have changed the format to: `file "filename.aof seq" 1 type b`